Frameworks/CoreGraphics/CGFont.mm

```diff
@@ -16,615 +16,134 @@
-            state->lineLen = &state->chars[state->curIndex] - state->lineStart;
-            if (state->lineLen < 0)
-                state->lineLen = 0;
-
-            //  We have hit a hard linebreak, consume it
-            state->curIndex++;
-            state->hitLinebreak = true;
-
-            if (sizeOut) {
-                sizeOut->width = ((float)penX) / 64.0f;
-            }
-            break;
-        }
-
-        FT_Vector advance;
-        int curChar = state->chars[state->curIndex];
-
-        advance.x = 0;
-        advance.y = 0;
-        advance.x = getCharWidth(state->chars, opaque, state->curIndex);
-
-        if (curChar == ' ' || wrapChars) {
-            //  Soft linebreak possibility
-            lastPossibleBreakPos = state->curIndex;
-            lastPossibleBreakWidth = penX;
-        }
-        if (curChar != 13) {
-            penX += advance.x;
-        }
-
-        float curWidth;
-
-        curWidth = ((float)(penX)) / 64.0f;
-        if (curWidth > width && curChar != ' ') {
-            if (lastPossibleBreakPos != -1) {
-                //  We must now perform a soft break
-                state->lineLen = &state->chars[lastPossibleBreakPos] - state->lineStart;
-                state->curIndex = lastPossibleBreakPos + 1;
-
-                if (curChar == ' ') {
-                    //  Do a hard line break
-                    state->hitLinebreak = true;
-                }
-                if (sizeOut) {
-                    sizeOut->width = ((float)lastPossibleBreakWidth) / 64.0f;
-                }
-            } else {
-                //  Draw as many characters as we can
-                if (state->lineStart != &state->chars[state->curIndex]) {
-                    state->lineLen = &state->chars[state->curIndex] - state->lineStart;
-                    penX -= advance.x;
-
-                    if (sizeOut) {
-                        sizeOut->width = ((float)penX) / 64.0f;
-                    }
-                    break;
-                } else {
-                    //  Draw at least one char
-                    state->lineLen = 1;
-                    state->curIndex++;
-
-                    if (sizeOut) {
-                        sizeOut->width = ((float)penX) / 64.0f;
-                    }
-                    break;
-                }
-            }
-            break;
-        }
-
-        state->curIndex++;
-    }
-
-    //  Did not wrap
-    if (state->lineLen == -1) {
-        state->lineLen = &state->chars[state->count] - state->lineStart;
-        if (sizeOut) {
-            sizeOut->width = ((float)penX) / 64.0f;
-        }
-    }
-
-    return ret;
+static void __CGFontInit(CFTypeRef cf) {
+    CGFontRef font = (CGFontRef)cf;
+    // ComPtr needs to be manually new'd, to avoid calling Create functions to an incomplete out-pointer
+    struct __CGFont* mutableFont = const_cast<struct __CGFont*>(font);
+    new (std::addressof(mutableFont->_dwriteFontFace)) ComPtr<IDWriteFontFace>();
 }
```

Comment:
This is necessary for integration with CF while having C++ members.

---
In reply to: [84555028](https://github.com/Microsoft/WinObjC/pull/1145#discussion_r84555028) [](ancestors = 84555028,84415801,83340473)